### PR TITLE
Update version and changelog for core-kit v4.1.0

### PR DESCRIPTION
**Pull request by vigh-m**
**Description of changes:**
* Update `release-version` to `4.1.0`
* Update Changelog

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by rpkelly — APPROVED** [review]

**Review by cbgbt — APPROVED** [review]

**Comment by vigh-m** [comment]
⬆️ Push to add #311 to the Changelog

**Review by ytsssun — APPROVED** [review]

**Comment by ytsssun** [comment]
> ⬆️ Push to add #311 to the Changelog

Nit: Should we callout explicitly for the ssm-agent bump?

**Comment by vigh-m** [comment]
⬆️ Push to remove changes being pulled from release

**Comment by vigh-m** [comment]
> Nit: Should we callout explicitly for the ssm-agent bump?

@ytsssun I'm not sure. We haven't referenced it explicitly in the Changelog every time in the past AFAIK. Is there something specific about this change that we want to call out.

**Review by KCSesh — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,23 @@
+# v4.1.0 (2024-12-10)
+
+## OS Changes
+* Enable plugins and detailed EBS volume stats for `nvme-cli` ([#269])
+* Set `LoaderTimeInitUSec` and `LoaderTimeExecUSec` in GRUB ([#273])
+* Update `ecs-agent` to v1.88.0 ([#303])
+* Third party package updates ([#308], [#311])
```
Comment (arnaldo2792) [inline CHANGELOG.md]: Technically, the `ecs-agent` update counts as part of the third-party package updates.

**Comment by arnaldo2792** [comment]
> Nit: Should we callout explicitly for the ssm-agent bump?

No, we should group all package updates in one line

**Comment by vigh-m** [comment]
⬆️ Update changelog based on comments

**Review by KCSesh — APPROVED** [review]

**Pull request merged**